sdk/include/opentelemetry/sdk/trace/etw_data.h

```diff
@@ -0,0 +1,530 @@
+#pragma once
```

Comment:
Agree with moving out of trace _going forward in a separate PR_ . But there is a reason why it is in there: this PR was opened before Logging API was added. Eventually we need to refactor it into common.

Disagree with `-contrib` : this is official Linux Foundation -sponsored activity. `ETW` on Windows is akin to `syslog` on Linux : `standard platform facility`. It would be odd if OpenTelemetry does not support the `standard platform facility` as its output. As a reference, I'd like to mention the Docker project, which supports the following destinations as 1st class citizen destinations --- and ETW is in there as well :)

Driver | Description
-- | --
none | No logs are available for the container and docker logs does not return any output.
local | Logs are stored in a custom format designed for minimal overhead.
json-file | The logs are formatted as JSON. The default logging driver for Docker.
syslog | Writes logging messages to the syslog facility. The syslog daemon must be running on the host machine.
journald | Writes log messages to journald. The journald daemon must be running on the host machine.
gelf | Writes log messages to a Graylog Extended Log Format (GELF) endpoint such as Graylog or Logstash.
fluentd | Writes log messages to fluentd (forward input). The fluentd daemon must be running on the host machine.
awslogs | Writes log messages to Amazon CloudWatch Logs.
splunk | Writes log messages to splunk using the HTTP Event Collector.
etwlogs | Writes log messages as Event Tracing for Windows (ETW) events. Only available on Windows platforms.
gcplogs | Writes log messages to Google Cloud Platform (GCP) Logging.
logentries | Writes log messages to Rapid7 Logentries.



Right now we binary-pack the output in MsgPack format. But eventually we will add an additional option to allow for the output of "dynamic manifest" ETW, that is fully compatible with all existing ETW trace tooling, such as Windows Performance Analyzer / Recorder, xperf. etc.

The listener for the MsgPack format will be added to `contrib`, as the listener is just a reference example how to monitor for incoming ETW events.